servers/zms/src/main/java/com/yahoo/athenz/zms/provider/ProviderHostnameVerifier.java

```diff
@@ -0,0 +1,61 @@
+/*
+ *
+ *  * Copyright The Athenz Authors
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package com.yahoo.athenz.zms.provider;
+
+import com.yahoo.athenz.auth.util.Crypto;
+import org.eclipse.jetty.util.StringUtil;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLPeerUnverifiedException;
+import javax.net.ssl.SSLSession;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+
+public class ProviderHostnameVerifier implements HostnameVerifier {
+    private final ServiceProviderManager serviceProviderManager;
+
+    public ProviderHostnameVerifier(ServiceProviderManager serviceProviderManager) {
+
+        this.serviceProviderManager = serviceProviderManager;
+    }
+
+    @Override
+    public boolean verify(String hostname, SSLSession session) {
```

Comment:
This method requires that the caller is running athenz issued CA certs which is not the case. It could run both with an athenz issued certs or with a cert issued by a public ca (athenz issued ca is only a requirement for the ZTS instance endpoint). So the method should just do a standard validation against the hostname extracted from the uri and if that fails then try to fetch the service name and compare that.